shoop/campaigns/apps.py

```diff
@@ -50,6 +53,10 @@ class CampaignAppConfig(AppConfig):
 
     def ready(self):
         post_save.connect(
-            update_customers_groups,
+            update_groups_after_order,
             sender=Payment,
             dispatch_uid="contact_group_sales:update_customers_groups")
+        post_save.connect(
```

Comment:
Instead of using signal for this you should just override ``ContactGroupSalesRange`` save-method.